src/shared/extension-util.c

```diff
@@ -135,3 +136,20 @@ int parse_env_extension_hierarchies(char ***ret_hierarchies) {
         *ret_hierarchies = TAKE_PTR(l);
         return 0;
 }
+
+int extension_forbidden_content_validate(const char *root) {
```

Comment:
This doesn't matter too much, but this function name has the smell of double negation. Are we validating that the extension has forbidden contents? It'd be easier to say `extension_has_forbidden_content()` and then return true if does, and false if it does not.